### DISCUSSION TITLE
"User not authorized"

### DISCUSSION THREAD

**Discussion by ralph-payne** [community/community#18479] [Copilot Conversations] [answered]
Hey team,

Not sure if this is the correct forum to be raising this, but I've been using co-pilot for the past 3-4 months and it's been working great and now I'm getting this error:
GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

[INFO] [auth] [2022-06-13T09:44:50.044Z] Invalid copilot token: missing token: 401 
[ERROR] [default] [2022-06-13T09:44:50.044Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

GitHub Copilot
v1.28.6134

Macbook Pro
Apple M1

Is there any reason this might have happened?
When I try to join the waitlist again, I get this message:
GitHub Copilot is already enabled on your account! [Install GitHub Copilot](https://github.com/github/copilot-docs) now.

Thanks in advance.

**Comment by yzAlvin** [comment]
It could be your token had expired

**Reply by ralph-payne** [reply]
thanks this response. Do you know how I can generate a new token?

**Comment by number0x0001** [comment]
I've heard of people fixed it by signing out of GitHub in VSCode, uninstalling Copilot, then signing back into GitHub and re-installing Copilot. Pretty annoying process, but I'd suggest trying it to see if it works for you.

**Reply by ralph-payne** [reply]
Many thanks, that's worked :)

**Comment by lethetai2605** [comment, marked as answer]
1. Try signing out of your GitHub account from your IDE.
2. Disable and re-enable the Copilot extension.

**Reply by ralph-payne** [reply]
Many thanks, that's worked :)

**Comment by sammcj** [comment]
FYI I've had this several times recently, it appears to be a bug in the VSCode extension, simply signing out and back in did not work. This is what fixed it for me:

1. Sign out of all Github in VSCode (Click profile icon, sign out)
2. Remove all Github extensions (Copilot, Copilot chat, Actions, Pull requests)
3. Completely close all VSCode sessions (ensure there's no processes running)
4. Open VSCode, reinstall extensions, sign back in to all of them
